### ISSUE TITLE
`use` method is error prone in its current format

### ISSUE THREAD

**Issue by paulbrie** [teleporthq/teleport-lib-js#8]
The use of `use` method is error prone: if you forget to call with wrapped with `async/await` the libraries are not loaded properly.

**Proposal**: remove `readPluginDefinitionFromUrl` from it and make it sync. Keep `readPluginDefinitionFromUrl` as a standalone method.

**Comment by nicutimofte** [comment]
Another option would be to have both `use` and `useAsync` methods for sync and async operations.

**Comment by crissmoldovan** [comment]
i agree with removing the possibility of using `use` with urls, but keep it for `useAsync`

**Comment by nicutimofte** [comment]
@paulbrie, are the discussed APIs already published?

**Comment by paulbrie** [comment]
Not yet. If we all agree on the API updates, we could move forward. @nicutimofte, do you want to take the lead on this?
